spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -118,7 +119,17 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] {
           return withInfos(scanExec, fallbackReasons.toSet)
         }
 
-        val typeChecker = new CometScanTypeChecker(scanImpl)
+        val possibleDefaultValues = getExistenceDefaultValues(scanExec.requiredSchema)
+        if (possibleDefaultValues.exists(d => {
+            d != null && (d.isInstanceOf[ArrayBasedMapData] || d
+              .isInstanceOf[GenericInternalRow] || d.isInstanceOf[GenericArrayData])
```

Comment:
Is `GenericInternalRow` for structs?

Reply:
Yep.

Reply:
They're already Java-native types at this point for Spark's internal interface. We can't check SQL types.